sunflower-land/sunflower-land
# [FEAT] Gems

# Description

Introduces Gems into Sunflower Land. Replaces all occurences of BBs with Gems.

Allows players to exchange BBs for Gems at a 1:100 ratio. This modal will pop up if a player ever has a Block Buck in their inventory - this could catch cases where BBs are escrowed in Auctions or other cases they could reappear.

Smaller gem amounts will help us with upcoming initiatives. We can also reduce Gems required for the following instantly:

1. Digging
2. Fishing Lures
3. Expansions

<img width="550" alt="Screenshot 2024-09-25 at 3 19 30 PM" src="https://github.com/user-attachments/assets/f456a458-5aea-4152-a2c4-2b8c7bba79fd">
<img width="534" alt="Screenshot 2024-09-25 at 3 19 33 PM" src="https://github.com/user-attachments/assets/d301982e-14a5-46e3-8b40-6a83cec84748">

<img width="524" alt="Screenshot 2024-09-26 at 2 29 59 PM" src="https://github.com/user-attachments/assets/51fdb22f-b0fb-4d61-9911-bdfea44ac426">

## How to test?

Update your `.env` - `VITE_BUY_GEMS_CONTRACT=0x6C9DB9fdA8b99fD5c17827ADD1926ab6dE725786`

1. Connect to local API
2. Run the game, migration modal should pop up (if you have Block Bucks).
3. Test buying Gems
4. Test using Gems on Restocks, Auctions, Farm Hands, Digs, Lures etc
